lib/logstash/plugin_mixins/aws_config/v2.rb

```diff
@@ -0,0 +1,96 @@
+module LogStash::PluginMixins::AwsConfig::V2
+  # This method is called when someone includes this module
+  def self.included(base)
+    # Add these methods to the 'base' given.
+    base.extend(self)
+    base.setup_aws_config
+  end
+
+  public
+  def setup_aws_config
+    # The AWS Region
+    config :region, :validate => LogStash::PluginMixins::AwsConfig::REGIONS_ENDPOINT, :default => LogStash::PluginMixins::AwsConfig::US_EAST_1 
+
+    # This plugin uses the AWS SDK and supports several ways to get credentials, which will be tried in this order...
+    # 1. Static configuration, using `access_key_id` and `secret_access_key` params in logstash plugin config
+    # 2. External credentials file specified by `aws_credentials_file`
+    # 3. Environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
+    # 4. Environment variables `AMAZON_ACCESS_KEY_ID` and `AMAZON_SECRET_ACCESS_KEY`
+    # 5. IAM Instance Profile (available when running inside EC2)
+    config :access_key_id, :validate => :string
+
+    # The AWS Secret Access Key
+    config :secret_access_key, :validate => :string
+
+    # The AWS Session token for temprory credential
+    config :session_token, :validate => :string
+
+    # Should we require (true) or disable (false) using SSL for communicating with the AWS API
+    # The AWS SDK for Ruby defaults to SSL so we preserve that
+    config :use_ssl, :validate => :boolean, :default => true, :deprecated => "This plugin uses `AWS-SDK` v2 which enforce SSL mode"
```

Comment:
Except from this configuration option, everything is the same for v1 and v2. Does it makes sense to have it in one place? what do you think?